File: 0345-reverse-vowels-of-a-string/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        a="aeiouAEIOU"
        l1=[]
        s1=""
        s2=""
        for i in s:
            if i in a:
                l1.append(i)
                s1=s1+"_"
                continue
            s1=s1+i
        l1.reverse()
        for i in s1:
            if i=="_":
                s2=s2+l1[0]
                l1.remove(l1[0])
                continue
            s2=s2+i
        return s2
        




        

File: 0345-reverse-vowels-of-a-string/solution.java
class Solution {
    public String reverseVowels(String s) {
        int i=0,j=s.length()-1;
        StringBuilder str=new StringBuilder(s);
        String v="AEIOUaeiou";
        while(i<j){
            char val1=s.charAt(i);
            if(v.indexOf(val1)==-1){
                i++;
                continue;
            }
            char val2=s.charAt(j);
            if(v.indexOf(val2)==-1){
                j--;
                continue;
            }
            str.setCharAt(i,val2);
            str.setCharAt(j,val1);
            i++;
            j--;
        }
        return str.toString();
    }
}
